src/lib.rs

```diff
@@ -561,25 +540,54 @@ impl TransClient {
     /// # Errors
     ///
     /// Any IO Error or Deserialization error
-    async fn call<RS>(&self, request: RpcRequest) -> Result<RpcResponse<RS>>
+    async fn call<RS>(&mut self, request: RpcRequest) -> Result<RpcResponse<RS>>
         where
             RS: RpcResponseArgument + DeserializeOwned + std::fmt::Debug,
     {
-        info!("Loaded auth: {:?}", &self.auth);
-        let rq: reqwest::RequestBuilder = self
-            .rpc_request()
-            .header("X-Transmission-Session-Id", self.get_session_id().await)
-            .json(&request);
-        info!(
-            "Request body: {:?}",
-            rq.try_clone()
-                .expect("Unable to get the request body")
-                .body_string()?
-        );
-        let resp: reqwest::Response = rq.send().await?;
-        let rpc_response: RpcResponse<RS> = resp.json().await?;
-        info!("Response body: {:#?}", rpc_response);
-        Ok(rpc_response)
+        let mut remaining_retry = 5; // TODO Config
+        loop {
+            if remaining_retry <= 0 {
+                error!("Out of retries!")
+                //break Err { Box::new("")??? }; // FIXME
```

Comment:
How could I return error here?

Reply:
Try to return an `Err(...)` with error object

Reply:
So we have to define an error struct then?